src/Discord.Net.Core/Entities/Interactions/Modals/ModalBuilder.cs

```diff
@@ -108,21 +108,100 @@ public ModalBuilder AddComponents(List<IMessageComponent> components, int row)
+        /// </exception>
+        public TMessageComponent GetComponent<TMessageComponent>(string customId) 
+            where TMessageComponent : class, IMessageComponent 
+            => Components.ActionRows?.SelectMany(r => r.Components.OfType<TMessageComponent>()).FirstOrDefault(c => c?.CustomId == customId)
+                ?? throw new ArgumentException($"There is no component of type {typeof(TMessageComponent).Name} with the specified custom ID in this modal builder.", nameof(customId));
+
+        /// <summary>
+        ///     Updates a <see cref="TextInputComponent"/> by the specified <paramref name="customId"/>.
+        /// </summary>
+        /// <param name="customId">The <see cref="TextInputComponent.CustomId"/> of the input to update.</param>
+        /// <param name="updateTextInput">An action that configures the updated text input.</param>
+        /// <returns>The current builder.</returns>
+        public ModalBuilder UpdateTextInput(string customId, Action<TextInputBuilder> updateTextInput)
+        {
+            var component = GetComponent<TextInputComponent>(customId);
+            var row = Components.ActionRows.First(r => r.Components.Contains(component));
+
+            var builder = new TextInputBuilder
+            {
+                Label = component.Label,
+                CustomId = component.CustomId,
+                Style = component.Style,
+                Placeholder = component.Placeholder,
+                MinLength = component.MinLength,
+                MaxLength = component.MaxLength,
+                Required = component.Required,
+                Value = component.Value
+            };
+
+            updateTextInput(builder);
+
+            row.Components.Remove(component);
+            row.AddComponent(builder.Build());
+
+            return this;
+        }
+        
+        /// <summary>
+        /// Updates the value of a <see cref="TextInputComponent"/> by the specified <paramref name="customId"/>.
+        /// </summary>
+        /// <param name="customId">The <see cref="TextInputComponent.CustomId"/> of the input to update.</param>
+        /// <param name="value">The new value to put.</param>
+        /// <returns>The current builder.</returns>
+        public ModalBuilder UpdateTextInput(string customId, object value)
+        {
+            UpdateTextInput(customId, x => x.Value = value.ToString());
+            return this;
+        }
+
+        /// <summary>
+        ///     Removes a component from this builder by the specified <paramref name="customId"/>.
+        /// </summary>
+        /// <param name="customId">The <see cref="MessageComponent.CustomId"/> of the component to remove.</param>
+        /// <returns>The current builder.</returns>
+        public ModalBuilder RemoveComponent(string customId)
+        {
+            Components.ActionRows.ForEach(r => r.Components.RemoveAll(c => c.CustomId == customId));
```

Comment:
This also should have a `null` check

Reply:
https://github.com/discord-net/Discord.Net/pull/2773/commits/67ee77de98e14662f051aa2d63ac50939a37d97b

Reply:
null check for `customId` is still missing.
use `Preconditions.NotNull`

Reply:
should this be done in all methods with the `customId` parameter?

Reply:
in some part https://github.com/discord-net/Discord.Net/pull/2644/commits/b1c70cd8f43045da4690ee208541aa703242cb41#diff-bee36ec4792aecd0f304bb053384a6db2c7faf50443b8d143b 66e212c7af474c inspired me to create these methods, but there is no such thing, although the functionality is not much different